src/toil/job.py

```diff
@@ -1654,3 +1757,29 @@ def promisedJobReturnValueUnpickleFunction(jobStoreString, jobStoreFileID):
 copy_reg.pickle(PromisedJobReturnValue,
                 promisedJobReturnValuePickleFunction,
                 promisedJobReturnValueUnpickleFunction)
+
+
+class PromisedRequirement(object):
+    def __init__(self, *args):
+        """
+        Class for generating function resource requirement values using promised return values.
+        (see toil.job.PromisedJobReturnValue)
+        :param args[0]: function for calculating job requirement
+                        default: lambda x: x
+        :param args[1:]: list of args, ordered with respect to function parameters
+        """
+        if hasattr(args[0], '__call__'):
```

Comment:
This catches `disk=PromisedReturn(child.rv())`. What about `disk=child.rv()`. Is that handled somewhere else?